### PR TITLE
Time dynamic geo json

### PR DESCRIPTION
**Pull request by ocefpaf**
Same as #736 with some minor fixes.

Ping @halfdanrump

### CODE REVIEW

File: folium/plugins/time_slider_choropleth.py [hunk 1]
```diff
@@ -1,69 +1,30 @@
...
-    ...                             x['properties']['name']=='Alabama' else
-    ...                             '#00ff00'}
-    >>> GeoJson(geojson, style_function=style_function)
-
-    """
-    def __init__(self, data, styledict, **kwargs):
-        super(TimeDynamicGeoJson, self).__init__(data, **kwargs)
-        assert isinstance(styledict, dict), 'styledict must be a dictionary'
+class TimeSliderChoropleth(GeoJson):
+    def __init__(self, data, styledict, name=None, overlay=True, control=True, **kwargs):
```
Comment (ocefpaf) [inline folium/plugins/time_slider_choropleth.py]: Still need some docs.

File: folium/plugins/time_slider_choropleth.py [hunk 2]
```diff
@@ -1,69 +1,30 @@
...
+        super(TimeSliderChoropleth, self).__init__(data, name=name, overlay=overlay, control=control)
+        if not isinstance(styledict, dict):
+            raise ValueError('styledict must be a dictionary, got {!r}'.format(styledict))
         for val in styledict.values():
-            assert isinstance(val, dict), 'each item in styledict must be a dictionary'
+            if not isinstance(val, dict):
+                raise ValueError('Each item in styledict must be a dictionary, got {!r}'.format(val))
```
Comment (ocefpaf) [inline folium/plugins/time_slider_choropleth.py]: `raise` instead of `assert`.

File: folium/plugins/time_slider_choropleth.py [hunk 3]
```diff
@@ -1,69 +1,30 @@
...
+            timestamps.update(set(feature.keys()))
+        timestamps = sorted(list(timestamps))
 
-        # make set of timestamps
-        self.timestamps = set()
-        for feature in self.styledict.values():
-            self.timestamps.update(set(feature.keys()))
-        self.timestamps = sorted(list(self.timestamps))
+        self.timestamps = json.dumps(timestamps)
+        self.styledict = json.dumps(styledict, sort_keys=True, indent=2)
```
Comment (ocefpaf) [inline folium/plugins/time_slider_choropleth.py]: By dumping `json` strings we avoid the `u''` in the code and fix the py27 failures.

File: tests/plugins/test_time_slider_choropleth.py [hunk 4]
```diff
@@ -51,23 +61,21 @@ def norm(col):
...
-    # We verify that imports
-    assert '<script src="https://d3js.org/d3.v4.min.js' in out  # noqa
-
-    # We verify that data has been inserted currectly
-    expected_timestamps = ("var timestamps = ['1454166000', '1456671600', '1459350000', "
-                           "'1461942000', '1464620400', '1467212400', '1469890800', "
-                           "'1472569200', '1475161200', '1477839600'];")
-    #assert expected_timestamps in out
+    # We verify that data has been inserted correctly
+    expected_timestamps = """var timestamps = ["1454198400", "1456704000", "1459382400"];"""  # noqa
```
Comment (ocefpaf) [inline tests/plugins/test_time_slider_choropleth.py]: I reduced this to make it easier to compare.
Not to self: these values are not the same on Travis-CI and locally. I need to double check this!

File: folium/plugins/time_slider_choropleth.py (hunk 3 above)
Comment (halfdanrump) [reply folium/plugins/time_slider_choropleth.py]: nice

**Comment by halfdanrump** [comment]
@ocefpaf I see a lot of people are submitting PRs, so you're probably pretty swamped :). Is there anything I can do to wrap up this PR?

Cheers,
Halfdan

**Comment by ocefpaf** [comment]
I'm traveling at the moment but I'll tackle this one first once I get back.

**Comment by ocefpaf** [comment]
@halfdanrump I am merging this but we could use a re-write of the docs later.

Thanks for the PR!

**Comment by halfdanrump** [comment]
@ocefpaf Got it, I'll add docs after the PR is complete and then issue a new PR :). Enjoy your trip!

**Pull request merged**